base_layer/chat_ffi/src/callback_handler.rs

```diff
@@ -103,8 +170,14 @@ impl CallbackHandler {
             "Calling ContactStatusChanged callback function for contact {}",
             data.address(),
         );
-        unsafe {
-            (self.callback_contact_status_change)(Box::into_raw(Box::new(data)));
+
+        match ChatFFIContactsLivenessData::try_from(data) {
+            Ok(data) => unsafe {
+                (self.callback_contact_status_change)(Box::into_raw(Box::new(data)));
```

Comment:
Why are we using a box here? Is the data to big tobe on the stack?
Box per say is not a problem, but using `Box::into_raw` means we have to ensure the safety of the data. Are we making sure we clean this up after we use it?

Reply:
We're using box because the data is accessed directly outside "our" scope, through the ffi boundaries. So we need to ensure it lives long enough. 

Currently, we are not ensuring it's cleaned up. So I'll make a destroy available.